### ISSUE TITLE
index_all.php not working

### ISSUE THREAD

**Issue by mikebywater** [gbirke/dokuwiki_solr#2]
Hello I am struggling to get the index_all.php to run correctly it is generating the following error.

Is it something you are familiar with?

PHP Warning:  simplexml_load_string(): Entity: line 1: parser error : Opening and ending tag mismatch: HR line 1 and body in /var/lib/dokuwiki/lib/plugins/solr/index_all.php on line 110
PHP Warning:  simplexml_load_string(): nt').</u></p><HR size="1" noshade="noshade"><h3>Apache Tomcat/6.0.35</h3></body> in /var/lib/dokuwiki/lib/plugins/solr/index_all.php on line 110
PHP Warning:  simplexml_load_string():                                                                                ^ in /var/lib/dokuwiki/lib/plugins/solr/index_all.php on line 110
PHP Warning:  simplexml_load_string(): Entity: line 1: parser error : Opening and ending tag mismatch: HR line 1 and html in /var/lib/dokuwiki/lib/plugins/solr/index_all.php on line 110
PHP Warning:  simplexml_load_string(): u></p><HR size="1" noshade="noshade"><h3>Apache Tomcat/6.0.35</h3></body></html> in /var/lib/dokuwiki/lib/plugins/solr/index_all.php on line 110
PHP Warning:  simplexml_load_string():                                                                                ^ in /var/lib/dokuwiki/lib/plugins/solr/index_all.php on line 110
PHP Warning:  simplexml_load_string(): Entity: line 1: parser error : Premature end of data in tag body line 1 in /var/lib/dokuwiki/lib/plugins/solr/index_all.php on line 110
PHP Warning:  simplexml_load_string(): u></p><HR size="1" noshade="noshade"><h3>Apache Tomcat/6.0.35</h3></body></html> in /var/lib/dokuwiki/lib/plugins/solr/index_all.php on line 110
PHP Warning:  simplexml_load_string():                                                                                ^ in /var/lib/dokuwiki/lib/plugins/solr/index_all.php on line 110
PHP Warning:  simplexml_load_string(): Entity: line 1: parser error : Premature end of data in tag html line 1 in /var/lib/dokuwiki/lib/plugins/solr/index_all.php on line 110
PHP Warning:  simplexml_load_string(): u></p><HR size="1" noshade="noshade"><h3>Apache Tomcat/6.0.35</h3></body></html> in /var/lib/dokuwiki/lib/plugins/solr/index_all.php on line 110
PHP Warning:  simplexml_load_string():

**Comment by gbirke** [comment]
This occurs when Solr returns a HTML page instead of the expected XML. This happens when Solr returns an error. The Solr behavior will be fixed in Solr 4.0. See [Issue SOLR-141](https://issues.apache.org/jira/browse/SOLR-141).

Try replacing line 110 ( `$xml = simplexml_load_string($result);` with the following:

``` php
if(!($xml = @simplexml_load_string($result))) {
    echo "Error parsing the Solr result document:\n$result\n";
}
```

This way you'll see what the Solr server sends back and can act accordingly.

**Comment by gbirke** [comment]
Instead of using the "bad" error suppression operator the new version I've pushed to the "refactoring" branch now catches HTML output and returns proper error messages from Solr.
